ui/components/app/permission-page-container/permission-page-container.component.js

```diff
@@ -103,6 +104,17 @@ export default class PermissionPageContainer extends Component {
 
     if (Object.keys(request.permissions).length > 0) {
       approvePermissionsRequest(request);
+      const searchParams = new URLSearchParams(
+        this.props.history.location.search,
+      );
+      const redirect = searchParams.get('redirect');
+      if (redirect) {
+        global.platform.openTab({
+          url: `${redirect}?${request.approvedAccounts
```

Comment:
@meppsilon What is the reason for including the approved accounts in the URL? They should be retrievable via the provider api (let me know if you need some guideance on how to do that). Is there some need this meets that the provider api does not?

Reply:
You know what, I don't think we need this after all. I think you are right. Thanks for pointing this out. The reason I put this here is because on entering the dapp, we only automatically connect the user's active address. I thought this could be a way to retrieve all of the approved addresses but I realize that we could also call to retrieve all of the addresses on entering the dapp as well and this wouldn't be necessary. We would, however, need a way to indicate that the user is coming from metamask because we want to show a modal but we could use a different search param for that.

Reply:
Updated